src/miner.cpp

```diff
@@ -104,13 +104,6 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
     if (Params().MineBlocksOnDemand())
         pblock->nVersion = GetArg("-blockversion", pblock->nVersion);
 
-    // Create coinbase tx
-    CMutableTransaction txNew;
-    txNew.vin.resize(1);
-    txNew.vin[0].prevout.SetNull();
-    txNew.vout.resize(1);
-    txNew.vout[0].scriptPubKey = scriptPubKeyIn;
-
     // Add dummy coinbase tx as first transaction
```

Comment:
Why does the old code say:

> CMutableTransaction txNew
> …
> pblock->vtx.push_back(CTransaction());

-instead of:

> CMutableTransaction txNew
> …
> pblock->vtx.push_back(txNew);

As this diff shows, ``txNew`` is not referenced until much later. Are those two different transactions?